File: app/src/main/java/com/example/leeeyou/zhihuribao/di/component/StoryComponent.java
package com.example.leeeyou.zhihuribao.di.component;

import com.example.leeeyou.zhihuribao.di.module.StoryModule;
import com.example.leeeyou.zhihuribao.view.activity.StoryActivity;
import com.example.leeeyou.zhihuribao.view.activity.StoryDetailActivity;
import com.example.leeeyou.zhihuribao.view.activity.fragment.StoryFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = StoryModule.class)
public interface StoryComponent {
    void inject(StoryActivity storyActivity);

    void inject(StoryDetailActivity storyDetailActivity);

    void inject(StoryFragment storyFragment);
}

File: app/src/main/java/com/xyz/leeeyou/zhihuribao/di/component/StoryComponent.kt
package com.xyz.leeeyou.zhihuribao.di.component

import com.xyz.leeeyou.zhihuribao.di.module.StoryModule
import com.xyz.leeeyou.zhihuribao.view.activity.StoryDetailActivity
import com.xyz.leeeyou.zhihuribao.view.fragment.StoryFragment
import dagger.Component
import javax.inject.Singleton

/**
 * ClassName:   StoryComponent
 * Description: [ Story ] module Component
 *
 * Author:      leeeyou
 * Date:        2018/2/24 15:15
 */
@Singleton
@Component(modules = [StoryModule::class])
interface StoryComponent {
    fun inject(storyDetailActivity: StoryDetailActivity)

    fun inject(storyFragment: StoryFragment)
}